kraken/namespace_actions/common_namespace_functions.py

```diff
@@ -19,48 +19,61 @@ def run(scenarios_list, config, wait_duration, failed_post_scenarios, kubeconfig
             scenario_config_yaml = yaml.full_load(f)
             for scenario in scenario_config_yaml["scenarios"]:
                 scenario_namespace = scenario.get("namespace", "^.*$")
-                scenario_label = scenario.get("label_selector", None)
+                scenario_label = scenario.get("label_selector", "")
+                if scenario_namespace != "^.*$":
+                    if scenario_label.strip() != "":
+                        logging.error("You can only have namespace or label set in your namespace scenario")
+                        logging.error(
+                            "Please set either namespace to '^.*$' or label_selector to blank ('') to continue"
+                        )
+                        sys.exit(1)
+                delete_count = scenario.get("delete_count", 1)
                 run_count = scenario.get("runs", 1)
                 namespace_action = scenario.get("action", "delete")
                 run_sleep = scenario.get("sleep", 10)
                 wait_time = scenario.get("wait_time", 30)
                 killed_namespaces = []
-                namespaces = kubecli.check_namespaces([scenario_namespace], scenario_label)
                 start_time = int(time.time())
                 for i in range(run_count):
-                    if len(namespaces) == 0:
-                        logging.error(
-                            "Couldn't %s %s namespaces, not enough namespaces matching %s with label %s"
-                            % (namespace_action, str(run_count), scenario_namespace, str(scenario_label))
-                        )
-                        sys.exit(1)
-                    selected_namespace = namespaces[random.randint(0, len(namespaces) - 1)]
-                    killed_namespaces.append(selected_namespace)
-                    try:
-                        runcommand.invoke("oc %s project %s" % (namespace_action, selected_namespace))
-                        logging.info(namespace_action + " on namespace " + str(selected_namespace) + " was successful")
-                    except Exception as e:
-                        logging.info(
-                            namespace_action + " on namespace " + str(selected_namespace) + " was unsuccessful"
-                        )
-                        logging.info("Namespace action error: " + str(e))
-                        sys.exit(1)
-                    namespaces.remove(selected_namespace)
-                    logging.info("Waiting %s seconds between namespace deletions" % str(run_sleep))
-                    time.sleep(run_sleep)
+                    logging.info("run count" + str(i))
+                    namespaces = kubecli.check_namespaces([scenario_namespace], scenario_label)
+                    for j in range(delete_count):
+                        logging.info("delete count" + str(j))
```

Comment:
Maybe change the format to "delete count: " to improve readability? Thoughts?

Reply:
didn't mean to keep that in there, thoughts on just deleting that line and line 38?

Reply:
Sure, should be print the config to help understand what values were passed?

Reply:
Updated